datafusion/physical-plan/src/joins/sort_merge_join.rs

```diff
@@ -1613,10 +1611,10 @@ impl SortMergeJoinStream {
 
             self.output_record_batches
                 .filter_mask
-                .extend(&BooleanArray::from(vec![None; num_rows]));
+                .append_nulls(num_rows);
             self.output_record_batches
                 .row_indices
-                .extend(&UInt64Array::from(vec![None; num_rows]));
+                .append_nulls(num_rows);
             self.output_record_batches
                 .batch_ids
                 .extend(vec![0; num_rows]);
```

Comment:
Btw this could avoid some allocation as well (using `resize` rather than extend based on `Vec`)

Reply:
good point, I'll check if there are other use cases we can modify

Reply:
https://github.com/apache/datafusion/pull/13861